test/generate_property_json.js

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env node
+
+const faker = require('faker')
+
+const nbInitOps = 16
+const nbRunOps = 64
+const specialChars = [':', '_', 'é', ' ', '%', '\\', '&', '@', 'É', 'Ç']
+
+let knownPaths = []
+let deletedPaths = []
+
+function similarPath () {
+  let p = faker.random.arrayElement(knownPaths)
+  let q = p.replace('é', '\u00E9')
+  if (p !== q) {
+    return q
+  }
+  q = p.toLowerCase()
+  if (p !== q) {
+    return q
+  }
+  q = p.toUpperCase()
+  if (p !== q) {
+    return q
+  }
+  return p + faker.random.arrayElement(specialChars)
+}
+
+function newPath () {
+  if (knownPaths.length > 0 && faker.random.number(4) === 0) {
+    return similarPath()
+  }
+  let p = faker.system.fileName()
+  if (faker.random.number(4) === 0) {
+    p = faker.random.arrayElement(specialChars) + p
+  }
+  if (knownPaths.length > 0 && faker.random.number(3) > 0) {
+    p = faker.random.arrayElement(knownPaths) + '/' + p
+  }
+  knownPaths.push(p)
+  return p
+}
+
+function knownPath () {
+  if (knownPaths.length === 0) {
+    return newPath()
+  }
+  return faker.random.arrayElement(knownPaths)
+}
+
+function deletedPath () {
+  if (deletedPaths.length === 0) {
+    return newPath()
+  }
+  return faker.random.arrayElement(deletedPaths)
+}
+
+function fileSize () {
+  return faker.random.number(32) + 2 ** (1 + faker.random.number(20))
+}
+
+function createNewDir () {
+  return { op: 'mkdir', path: newPath() }
+}
+
+function createNewFile () {
+  return { op: 'create_file', path: newPath(), size: fileSize() }
+}
+
+function recreateDeletedDir () {
+  return { op: 'mkdir', path: deletedPath() }
+}
+
+function recreateDeletedFile () {
+  return { op: 'create_file', path: deletedPath() }
+}
+
+function updateFile () {
+  return { op: 'update_file', path: knownPath(), size: fileSize() }
+}
+
+function mvToNewPath () {
+  let p = knownPath()
+  deletedPaths.push(p)
+  return { op: 'mv', from: p, to: newPath() }
+}
+
+function mvToDeletedPath () {
+  let p = knownPath()
+  let to = deletedPath()
+  deletedPaths.push(p)
+  return { op: 'mv', from: p, to: to }
+}
+
+function rm () {
+  let p = knownPath()
+  deletedPaths.push(p)
+  return { op: 'rm', path: p }
+}
+
+function sleep () {
+  const s = 2 ** (1 + faker.random.number(3))
```

Comment:
It will evolve. For the moment, with our current watcher, sleep with less than 2 seconds was not very interesting. But if we improve the watcher, having smaller sleeps will be necessary to test more cases.